### PR TITLE
improve diagnostics when registry configuration is broken

### PR DESCRIPTION
**Pull request by tomerd**
motivation: better user experience

changes: catch the JSON error and wrap it with a more detailed diagnostics

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@yim-lee

**Review by yim-lee — APPROVED** [review]

File: Sources/Workspace/Workspace+Configuration.swift [hunk 1]
```diff
@@ -684,9 +684,13 @@ extension Workspace.Configuration {
...
 
-            let data: Data = try fileSystem.readFileContents(self.path)
-            let decoder = JSONDecoder.makeWithDefaults()
-            return try decoder.decode(RegistryConfiguration.self, from: data)
+            do {
+                let data: Data = try fileSystem.readFileContents(self.path)
+                let decoder = JSONDecoder.makeWithDefaults()
+                return try decoder.decode(RegistryConfiguration.self, from: data)
+            } catch {
+                throw StringError("Failed loading registries configuration from '\(self.path)': \(error)")
```
Comment (MaxDesiatov) [inline Sources/Workspace/Workspace+Configuration.swift:692]: nit: in other places we use backquotes, not sure which of these quotes do we want to apply everywhere for consistency?

```suggestion
                throw StringError("Failed loading registries configuration from `\(self.path)`: \(error)")
```

File: Sources/Workspace/Workspace+Configuration.swift [hunk 2]
```diff
@@ -684,9 +684,13 @@ extension Workspace.Configuration {
                 return RegistryConfiguration()
             }
 
-            let data: Data = try fileSystem.readFileContents(self.path)
-            let decoder = JSONDecoder.makeWithDefaults()
-            return try decoder.decode(RegistryConfiguration.self, from: data)
+            do {
+                let data: Data = try fileSystem.readFileContents(self.path)
+                let decoder = JSONDecoder.makeWithDefaults()
+                return try decoder.decode(RegistryConfiguration.self, from: data)
```
Comment (MaxDesiatov) [inline Sources/Workspace/Workspace+Configuration.swift:690]: nit: we have a helper function that could make this slightly shorter
```suggestion
                let decoder = JSONDecoder.makeWithDefaults()
                return try decoder.decode(path: self.path, fileSystem: self.fileSystem, as: RegistryConfiguration.self)
```

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci smoke test windows

**Pull request merged**